### PR TITLE
Add Arctic region aggregation for ocean and sea ice

### PR DESCRIPTION
**Pull request by xylar**
This merge adds aggregation functions for Arctic regions that are relevant to both ocean and sea-ice.

I also included clean-up in 2 places I discovered along the way, removing one irrelevant comment and fixing a typo.

### CODE REVIEW

**Comment by xylar** [comment]
@milenaveneziani, thanks for helping to clear things up in your comment (https://github.com/MPAS-Dev/MPAS-Analysis/pull/756#issuecomment-735606985).  It seems like the ``arcticRegions20200621.geojson`` file I pointed to was actually created with my incorrect definitions of the Arctic regions and should not be used.

Please check to make sure the aggregation functions I added here make sense.  They're really simple, so the main thing would be to make sure the list of regions they produce (see the modifications to the documentation) are what you expect.  Also, let me know if you're okay with having a `seaice` subdirectory under `aggregation` (even though the regions are currently in the `ocean`  folder in `geometric_data`).

File: docs/aggregation.rst [hunk 1]
```diff
@@ -37,6 +37,15 @@ Meridional Overturning Circulation Basins
 The function :py:func:`geometric_features.aggregation.ocean.moc` aggregates
 basins for computing the MOC: Atlantic, IndoPacific, Pacific and Indian
 
+Arctic Regions
+~~~~~~~~~~~~~~
+
+The function :py:func:`geometric_features.aggregation.ocean.arctic` aggregates
+regions of the Arctic: Baffin Bay, Baltic Sea, Barents Sea, Canada Basin,
```
Comment (milenaveneziani) [inline docs/aggregation.rst:44]: Above, you may want to modify to `of the Arcic (and subArctic)`.

**Comment by milenaveneziani** [comment]
Thanks @xylar very much for doing this.

I just realized that, in my original `arcticRegions.geojson`, I also included the following regions:
1) `Beaufort Gyre`: subregion of the `Canada Basin` feature already included here, which is helpful when doing comparisons with observational data of freshwater content (Proshutinsky's papers).
2) `Beaufort Gyre shelf`: complements 1), as the region in 1) does not include the shelf.
3) `Arctic (no Barents/Kara Seas)`: helpful as some papers exclude the Barents and Kara Seas when talking about the Arctic. I think at one point you asked me about changing the name of this region, as MPAS-Tools was having trouble with spaces and slashes. I am totally fine with that. Could you please remind me what name you ended up using?

So, I think we could add these 3 regions. The first 2 have tag `Arctic_Proshutinsky`. The third one does not have a tag, but it is aggregated from the following regions: `Chukchi Sea`, `Canada Basin`, `East Siberian Sea`, `Laptev Sea`, `Central Arctic`.

File: docs/aggregation.rst (hunk 1 above)
Comment (milenaveneziani) [reply docs/aggregation.rst:44]: Could add: 'Arctic Ocean (no Barents/Kara Seas)' (or related name), 'Beaufort Gyre', and 'Beaufort Gyre shelf'.

File: docs/api.rst [hunk 2]
```diff
@@ -95,3 +95,6 @@ Aggregation
    aggregation.ocean.antarctic
    aggregation.ocean.ice_shelves
    aggregation.ocean.transport
+
+   aggregation.seaice
+   aggregation.seaice.arctic
```
Comment (milenaveneziani) [inline docs/api.rst:100]: Nice. And yes, I like the idea of having the seaice folder with these regions in it.

File: geometric_features/aggregation/ocean/arctic_regions.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    -------
+    fc : geometric_features.FeatureCollection
+        The new feature collection with ocean-relevant Arctic regions
+    """
+    # Authors
+    # -------
+    # Milena Veneziani, Xylar Asay-Davis
+
+    fc = gf.read(componentName='ocean', objectType='region',
+                 tags=['Arctic'])
```
Comment (milenaveneziani) [inline geometric_features/aggregation/ocean/arctic_regions.py:20]: so basically here we would add the `Arctic_Proshutinsky` tag, and then add a few lines of code to aggregate the regions to make Arctic-noBarentsKaraSeas. I can do this if you like.

**Comment by xylar** [comment]
@milenaveneziani, thanks, that clears up some more of my confusion, since I had noticed the missing features.  I'll add your suggested changes.

**Comment by xylar** [comment]
The suggested name change was to `"Arctic Ocean without Barents and Kara Seas"`.  Something shorter would be nice but it can't have punctuation that isn't allowed in file names.

**Comment by milenaveneziani** [comment]
how about 'Arctic-noBarentsKaraSeas'?

**Comment by xylar** [comment]
That is more compact but I think the spaces do help.  I'm not sure if the dash is okay, probably so.

**Comment by xylar** [comment]
Let's see if `Arctic Ocean - no Barents, Kara Seas` will work.

**Comment by xylar** [comment]
The following:
``` python
from geometric_features.aggregation.ocean import arctic
from geometric_features import GeometricFeatures
arctic(GeometricFeatures()).to_geojson('arctic_ocean.geojson')
```
Produces a file with these names:
```
$ grep '"name": ' arctic_ocean.geojson 
                "name": "Baffin Bay",
                "name": "Baltic Sea",
                "name": "Barents Sea",
                "name": "Beaufort Gyre",
                "name": "Beaufort Gyre Shelf",
                "name": "Canada Basin",
                "name": "Canadian Archipelago",
                "name": "Central Arctic",
                "name": "Chukchi Sea",
                "name": "East Siberian Sea",
                "name": "Greenland Sea",
                "name": "Hudson Bay",
                "name": "Irminger Sea",
                "name": "Kara Sea",
                "name": "Labrador Sea",
                "name": "Laptev Sea",
                "name": "North Sea",
                "name": "Norwegian Sea",
                "name": "Arctic Ocean - no Barents, Kara Seas",
```
So far, so good.  I'm now going to rebuild the test package and test this out in MPAS-Analysis on Anvil.

**Comment by xylar** [comment]
@milenaveneziani, if the results at:
https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.xylar/analysis_testing/add_arctic_regions/arctic/ocean/index.html
look okay to you (given the very coarse resolution of the test simulation), I'll merge this and make another release so we can use this in MPAS-Analysis.  Let me know. There's no rush as far as I'm concerned, as long as you don't need this functionality in E3SM-Unified.

**Comment by milenaveneziani** [comment]
@xylar: that looks great! It will be nice to have that functionality in MPAS-Analysis. Thanks.

File: docs/aggregation.rst (hunk 1 above)
Comment (milenaveneziani) [reply docs/aggregation.rst:44]: @xylar: don't forget to modify this text above before merging.

**Review by milenaveneziani — APPROVED** [review]

File: docs/aggregation.rst [hunk 4]
```diff
@@ -37,6 +37,15 @@ Meridional Overturning Circulation Basins
...
+Canadian Archipelago, Central Arctic, Chukchi Sea, East Siberian Sea,
+Greenland Sea, Hudson Bay, Irminger Sea, Kara Sea, Labrador Sea, Laptev Sea,
+North Sea, Norwegian Sea.
```
Comment (xylar) [inline docs/aggregation.rst:47]:
```suggestion
The function :py:func:`geometric_features.aggregation.ocean.arctic` aggregates
regions of the Arctic: Baffin Bay, Baltic Sea, Barents Sea, Beaufort Gyre,
Beaufort Gyre Shelf, Canada Basin, Canadian Archipelago, Central Arctic, 
Chukchi Sea, East Siberian Sea, Greenland Sea, Hudson Bay, Irminger Sea, 
Kara Sea, Labrador Sea, Laptev Sea, North Sea, Norwegian Sea, and
Arctic Ocean - no Barents, Kara Seas.
```

File: docs/aggregation.rst (hunk 1 above)
Comment (xylar) [reply docs/aggregation.rst]: Thanks, I missed these suggestions somehow.

**Pull request merged**